sum_test.go

```diff
@@ -177,3 +180,46 @@ func TestBasicSum(t *testing.T) {
 		}
 	}
 }
+
+var sink []byte
+
+type codeNamePair struct {
+	id   uint64
+	name string
+}
+
+func BenchmarkSumAllLarge(b *testing.B) {
+	var data [1024 * 1024 * 16]byte
+	_, err := io.ReadFull(rand.Reader, data[:])
```

Comment:
`rand.Read` does the same thing. We probably shouldn't use cryptographic random here though, getting 16 MB from there might be slow. Better to use a `math/rand` reader.

Reply:
Fixed